### ISSUE TITLE
How to Convert MLVC-S Models to ONNX

### ISSUE THREAD

**Issue by TofuMars** [microsoft/mlvc#19]
I am having troubles converting the MLVC-S models with the video/convert.py script. I use an Ubuntu environment, and I am trying to convert the example [mlvc-s-perceptual-v1.ckpt](https://mlvideopub.blob.core.windows.net/mlvc/models/mlvc-s-perceptual-v1.ckpt) model into ONNX. I tried some modifications to the source code with AI assistance to see if it is possible, but I am facing some low accuracy issues with those workarounds. I am struggling with the following issues:

- the state_dict for mlvc-s-perceptual-v1.ckpt does not match model dmc61s, dmc61sr_lrelu, dmc61sbr_lrelu, or dmc61sbr_reglu. I always get a mismatch with the state_dict.
- There will be dimensional mismatches for the conversion when I do try to use the MLVC-S model. I included my model_configs.yaml file in my package that seems to get the dimensions correct, but I am not confident if it would create an accurate port of the model.
- Some of the biases in video/conversion/_exporter/_onnx_utils.py had to be disabled for my workaround to work. I commented out depth_to_space_crd_to_dcr and split_gated_conv to allow the script to proceed further.

Is there a way to properly convert the MLVC-S models accurately with the source code as-is? If I use the full model, I am able to convert it fine with the following uv command:

uv run convert.py export --model-version dmc61sbr_reglu --model-type onnx --weights-path /home/tofumars/mlvc-psnr-v1.ckpt --exporter-params-json "{\"test_video_path\":\"/home/tofumars/sample_640x368.yuv\",\"image_width\":640,\"image_height\":368}" --target-device generic

When I try to convert the MLVC-S model with my work arounds with the following command:
uv run convert.py export --precision fp16 --model-version dmc61sbr_reglu --weights-version v1_fused --model-type onnx --weights-path /home/tofumars/mlvc-s-psnr-v1.ckpt --exporter-params-json "{\"test_video_path\":\"/home/tofumars/sample_640x368.yuv\",\"image_width\":640,\"image_height\":368}" --target-device generic

I can only get a PSNR of 5dB, which seems to be poor.

My modifications and validation yuv file are attached here:

[09_08_2026_mlvcs_to_onnx_attempt.zip](https://github.com/user-attachments/files/31980434/09_08_2026_mlvcs_to_onnx_attempt.zip)

**Comment by TofuMars** [comment]
I want to add a note that regardless of mlvc-s model (whether it is the mlvc-s psnr model or perceptual model), that they experience the same issues when trying to convert them with the convert.py script

**Comment by TofuMars** [comment]
I am happy to say that as of https://github.com/microsoft/mlvc/commit/1425880bf679bbafdaffb7be3fb61f80448fb0b6
the MLVC-S model conversion is working for me!

If anyone wants to know the commands I used, I used the following:

uv run convert.py export --model-version dmc61sbr_mini_reglu --model-type onnx --weights-path /home/tofumars/mlvc-s-psnr-v1.ckpt --exporter-params-json "{\"test_video_path\":\"/home/tofumars/sample_640x368.yuv\",\"image_width\":640,\"image_height\":368}" --target-device generic

And I received the following result:

Runtime parameters:
+-------------------------+-------+
|         Common          | Value |
+-------------------------+-------+
|  CoreML Compute Units   |  npu  |
| CoreML Fast Prediction  | True  |
| ONNX Execution Provider |  cpu  |
|      Torch Device       |  cpu  |
+-------------------------+-------+

Validation results (QP=21):
  - video=/home/tofumars/sample_640x368.yuv
  - size=640x368, frames=48
+---------------+------+---------+------------+------------+--------+    +------------------+-------------+-----+-----------+
| Metric        | Unit | AzureML | Conversion | Validation | Delta  |    | Timer name       | Median (ms) | FPS | Share (%) |
+---------------+------+---------+------------+------------+--------+    +------------------+-------------+-----+-----------+
| PSNR (min)    |  dB  |   nan   |   31.16    |   31.16    | -0.00  |    | MLVCEncoderTotal |    135.1    | 7.4 |  50.5 %   |
| PSNR (max)    |  dB  |   nan   |   32.68    |   32.74    |  0.06  |    | MLVCDecoderTotal |    116.3    | 8.6 |  43.5 %   |
| PSNR (median) |  dB  |   nan   |   32.24    |   32.24    | -0.00  |    | MLVCEncoder      |    135.1    | 7.4 |  50.5 %   |
| PSNR (mean)   |  dB  |   nan   |   32.22    |   32.22    |  0.01  |    | MLVCDecoder      |    116.3    | 8.6 |  43.5 %   |
| PSNR Y (mean) |  dB  |   nan   |   33.09    |   33.09    |  0.00  |    | FrameLoopTotal   |    267.6    | 3.7 |  100.0 %  |
| PSNR U (mean) |  dB  |   nan   |   31.87    |   31.93    |  0.06  |    +------------------+-------------+-----+-----------+
| PSNR V (mean) |  dB  |   nan   |   27.36    |   27.32    | -0.04  |
| BPP (mean)    | bpp  |   nan   |   0.0157   |   0.0158   | 0.0001 |
+---------------+------+---------+------------+------------+--------+

Validation results (QP=63):
  - video=/home/tofumars/sample_640x368.yuv
  - size=640x368, frames=48
+---------------+------+---------+------------+------------+--------+    +------------------+-------------+-----+-----------+
| Metric        | Unit | AzureML | Conversion | Validation | Delta  |    | Timer name       | Median (ms) | FPS | Share (%) |
+---------------+------+---------+------------+------------+--------+    +------------------+-------------+-----+-----------+
| PSNR (min)    |  dB  |   nan   |   33.36    |   33.36    | -0.00  |    | MLVCEncoderTotal |    135.4    | 7.4 |  50.0 %   |
| PSNR (max)    |  dB  |   nan   |   35.84    |   35.84    |  0.01  |    | MLVCDecoderTotal |    118.2    | 8.5 |  43.6 %   |
| PSNR (median) |  dB  |   nan   |   35.11    |   35.11    |  0.00  |    | MLVCEncoder      |    135.4    | 7.4 |  50.0 %   |
| PSNR (mean)   |  dB  |   nan   |   35.09    |   35.09    | -0.00  |    | MLVCDecoder      |    118.2    | 8.5 |  43.6 %   |
| PSNR Y (mean) |  dB  |   nan   |   36.16    |   36.16    | -0.00  |    | FrameLoopTotal   |    270.8    | 3.7 |  100.0 %  |
| PSNR U (mean) |  dB  |   nan   |   34.20    |   34.19    | -0.00  |    +------------------+-------------+-----+-----------+
| PSNR V (mean) |  dB  |   nan   |   29.54    |   29.53    | -0.01  |
| BPP (mean)    | bpp  |   nan   |   0.1231   |   0.1234   | 0.0004 |
+---------------+------+---------+------------+------------+--------+
